core/state.py

```diff
@@ -2403,11 +2403,9 @@ def Get(self, name):
         First, we search for a proc, and then a sh-func. This means that procs
         can shadow the definition of sh-funcs.
         """
-        vars = self.mem.var_stack[0]
-        if name in vars:
-            maybe_proc = vars[name]
-            if maybe_proc.val.tag() == value_e.Proc:
-                return cast(value.Proc, maybe_proc.val)
+        maybe_proc = self.mem.GetValue(name)
```

Comment:
It is not good enough just to check `self.mem.var_stack[0]` for the `parseArgs` eval test. I think this may also fix issues like:

```
proc my-proc {
  var toCall = someOtherProc
  toCall foo bar baz
}
```

---

I should also look at GetNames because now it's doing something different

Reply:
This should be good now. We will leave `GetNames` as-is after the discussion here: https://oilshell.zulipchat.com/#narrow/stream/384942-language-design/topic/value.2EProc.20in.20the.20var.20namespace.